cloudevents/working-drafts/dds-protocol-binding.md

```diff
@@ -0,0 +1,265 @@
+wire protocol.
+DDS ensures reliable data delivery for latency-sensitive payloads,
+supporting Quality of Service (QoS) parameters like reliability, lifespan
+and partitioning. DDS is most commonly used for mission-critical applications
+in industries such as aerospace, healthcare, and industrial automation.
+
+This binding specification defines how attributes and data of a CloudEvent is
+mapped to a DDS message format.
+
+### 1.3. Content Modes
+
+The CloudEvents specification defines three content modes for transferring
+events: _structured_, _binary_ and _batch_. 
+
+Every compliant implementation of the CloudEvent specification SHOULD support
+both structured and binary modes. The DDS protocol binding supports both these modes.
+
+The DDS protocol binding does not currently support the _batch_ content mode.
+
+In the _structured_ content mode, event metadata attributes and event data are
+placed into the DDS message using an [event format](#14-event-formats).
+
+In the _binary_ content mode, the value of the event `data` is placed into
+the DDS message's `data` section as-is; all other event attributes are
+mapped to the DDS message's event metadata fields.
+
+### 1.4. Event Formats
+
+Event formats define how an event is expressed in a particular data format.
+All implementations of this specification that support the _structured_
+content mode MUST support the [JSON event format][json-format].
+
+The DDS protocol binding additionally supports the
+[DDS event format][dds-message-format].
+
+### 1.5. Security
+
+This specification does not introduce any new security features for DDS, or
+mandate specific existing features to be used.
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][ce] event
+attributes.
+
+### 2.1. data
+
+`data` is assumed to contain opaque application data that is encoded as declared
+by the `datacontenttype` attribute. 
+
+There are two values of `datacontenttype` currently supported:
+- `cloudevent/json` for JSON data
+- `application/cloudevent+dds` for binary and plain text data types
+
+If the `datacontenttype` attribute is set to `application/cloudevent+dds`, the
+`datacontentencoding` attribute defines the encoding of the message body. 
+
+The `content-type` field in the message header MUST be consistent with the
+`datacontenttype` attribute.
+
+## 3. DDS Message Mapping
+
+The content type is chosen by the sender of the event. The receiver of the
+event can distinguish between content modes by inspecting the `content-type`
+header of the DDS message.
+
+If the header is present and its value is `cloudevent/json`, the receiver
+decodes the message as JSON data.
+
+If the header is present and its value is `application/cloudevent+dds`, the
+receiver decodes the message into the DDS message format, using the
+`datacontentencoding` attribute to determine the encoding of the message body.
+There are three valid values of the `datacontentencoding` attribute
```

Comment:
3 or 2?

Reply:
It is 2! Another good catch - at first, I was thinking the JSON structured mode should go here too - but that is defined by its own format.